synapse/config/room_directory.py

```diff
@@ -20,7 +24,7 @@
 class RoomDirectoryConfig(Config):
     section = "roomdirectory"
 
-    def read_config(self, config, **kwargs):
```

Comment:
`config`? `kwargs` maybe?

Reply:
I have a separate branch which does all of the `read_config` and `generate_config_section` methods at once. Hope that's OK!

Reply:
aye that's reet :)